### DISCUSSION TITLE
Why compare partition id when selecting parents ?

### DISCUSSION THREAD

**Discussion by jinwon0325kim** [openthread/openthread#7598] [Q&A] [answered]
Hi All,

Looking at the code, check Partition Id when selecting parents.
However, Partition Id is created randomly and I don't understand how to use it to select parents.

The `HandleParentResponse` function performs the `ComparePartitions` fuction that compares the newly received parent information 
with the previously stored parent information.

```
if (aLeaderDataA.GetPartitionId() != aLeaderDataB.GetPartitionId())
{
    ExitNow(rval = aLeaderDataA.GetPartitionId() > aLeaderDataB.GetPartitionId() ? 1 : -1);
}
```

**Comment by jwhui** [comment, marked as answer]
Thread specifies how Thread devices merge partitions. Partitions with larger Partition ID values take precedence. The code you referenced implements the specified behavior. Merging partitions helps ensure that devices can communicate with each other when radio connectivity exists. See [Thread 1.1.1](https://www.threadgroup.org/support#specifications) Section 5.16.3 for more details.
